src/main/scala/io/flow/localization/rates/Localizer.scala

```diff
@@ -124,28 +125,48 @@ class LocalizerImpl @Inject() (dataClient: DataClient, rateProvider: RateProvide
     implicit executionContext: ExecutionContext
   ): Future[Option[FlowSkuPrice]] = {
     val key = CountryKey(country, itemNumber)
-    getPricing(key)
+    val defaultCurrency = Countries.mustFind(country).defaultCurrency
+    getPricing(key, defaultCurrency)
   }
 
   override def getSkuPricesByCountry(country: String, itemNumbers: Iterable[String])(
     implicit executionContext: ExecutionContext
   ): Future[List[Option[FlowSkuPrice]]] = {
     val keys = itemNumbers.map(CountryKey(country, _))
-    getPricings(keys)
+    val defaultCurrency = Countries.mustFind(country).defaultCurrency
+    getPricings(keys, defaultCurrency)
   }
 
-  private def getPricings(keyProviders: Iterable[KeyProvider])(
+  private def getPricings(keyProviders: Iterable[KeyProvider], targetCurrency: Option[String])(
     implicit executionContext: ExecutionContext
   ): Future[List[Option[FlowSkuPrice]]] = {
-    dataClient.mGet[Array[Byte]](keyProviders.map(_.getKey).toSeq).map { optionalPrices =>
-      optionalPrices.map(toFlowSkuPrice).toList
+    val futureOptionalPrices = dataClient.mGet[Array[Byte]](keyProviders.map(_.getKey).toSeq)
+
+    futureOptionalPrices.flatMap { optionalPrices =>
+      optionalPrices.map(toFlowSkuPrice) match {
+        case Nil => {
```

Comment:
Let's add a test for `getSkuPricesByCountry` since this condition will not work as expected.\

The size of `optionalPrices : Seq[Option[...]]`will always be the same as `keyProviders` and therefore will most likely never be empty.

We want to `mGet` all the undefined ones (and keep the defined ones?)

Reply:
Thought of this and am in the process of adding a test.  I actually think, the expectation is that is *will* always be empty when the country based key does not exist, and thus, will return all prices from the conversion.

Reply:
If not found, `Seq(1, 2, 3)` would be `Seq(None, None, None)`. It is critical that the size remains the same.

Reply:
Understood about what `mGet` returns.  Successfully wrote a failing test, my other assumption should still stand in that they should all be defined or undefined - but will take a look

Reply:
updated: https://github.com/flowcommerce/lib-localization/pull/38/files#diff-09f31c26647f6a0f37c8c6f56dc6cc2cR138
added test: https://github.com/flowcommerce/lib-localization/pull/38/files#diff-537ac046ca4816391ef2964bca1ffbbdR268

Reply:
They will not necessarily be all defined or undefined. It is still possible to have missing skus in the store for a given country. Do we then want to use the fallback case for all of them and discard the more correct ones?

Reply:
ack - as mentioned above, will take a look at this scenario/edge case

Reply:
added code to handle this scenario + test